core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -40,12 +40,22 @@ trait Queue[F[_], A] { self =>
   /** Dequeue one `A` from this queue. Completes once one is ready. */
   def dequeue1: F[A]
 
+  /** Dequeue all available `A`s from this queue. Completes once any values are ready. */
+  def dequeueAvailable1: F[Vector[A]]
+
   /** Like `dequeue1` but provides a way to cancel the dequeue. */
   def cancellableDequeue1: F[(F[A], F[Unit])]
 
+  /** Like `dequeueAvailable1` but provides a way to cancel the dequeue. */
+  def cancellableDequeueAvailable1: F[(F[Vector[A]], F[Unit])]
+
   /** Repeatedly call `dequeue1` forever. */
   def dequeue: Stream[F, A] = Stream.bracket(cancellableDequeue1)(d => Stream.eval(d._1), d => d._2).repeat
 
+  /** Repeatedly call `dequeueAvailable1` forever. */
+  def dequeueAvailable: Stream[F, A] =
```

Comment:
I just thought a little bit about `dequeueAvailable` and I think I would like it much more as 
```scala
def dequeueAvailable:Pipe[F, Int,A] = ???
```
the reason for this is the you may introduce very nice flow control based on some signal of `int` vaues that indicates how much client want to proceed at any given time. 

Also this is much general signature, and perhaps could allow us same code for dequeue and dequeueAvailable, where dequeue is really just special case for dequeueAvailable. If the performance will be an issue, we can always specialise dequeue, otherwise dequeue is for me just ```constant(1).through(dequeueAvailable)```

@pchiusano @fiadliel @mpilquist  any thoughts?